sdk/eventhub/eventhubs-checkpoint-blob/src/blobPartitionManager.ts

```diff
@@ -0,0 +1,209 @@
+   * @return A list partitions this instance successfully claimed ownership.
+   */
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    let partitionOwnershipArray: PartitionOwnership[] = [];
+    for (const ownership of partitionOwnership) {
+      const blobName = `${ownership.eventHubName}/${ownership.consumerGroupName}/${ownership.partitionId}`;
+      try {
+        let updatedBlobResponse;
+        const blobClient = this._containerClient.getBlobClient(blobName);
+        if (ownership.eTag) {
+          updatedBlobResponse = await blobClient.setMetadata(
+            {
+              OwnerId: ownership.ownerId ? ownership.ownerId : "",
+              SequenceNumber: ownership.sequenceNumber ? ownership.sequenceNumber.toString() : "",
+              Offset: ownership.offset ? ownership.offset.toString() : ""
+            },
+            {
+              blobAccessConditions: {
+                modifiedAccessConditions: {
+                  ifMatch: ownership.eTag
+                }
+              }
+            }
+          );
+        } else {
+          const blockBlobClient = blobClient.getBlockBlobClient();
+          updatedBlobResponse = await blockBlobClient.upload("", 0, {
+            metadata: {
+              OwnerId: ownership.ownerId,
+              SequenceNumber: ownership.sequenceNumber ? ownership.sequenceNumber.toString() : "",
+              Offset: ownership.offset ? ownership.offset.toString() : ""
+            },
+            accessConditions: {
+              modifiedAccessConditions: {
+                ifNoneMatch: "*"
+              }
+            }
+          });
+        }
+        if (updatedBlobResponse.lastModified) {
+          ownership.lastModifiedTimeInMS = Date.parse(
+            updatedBlobResponse.lastModified.toISOString()
+          );
+        }
+        ownership.eTag = updatedBlobResponse.eTag;
+        partitionOwnershipArray.push(ownership);
+        log.blobPartitionManager(
+          `[${ownership.ownerId}] Claimed ownership successfully for partition: ${ownership.partitionId}`,
+          `LastModifiedTime: ${ownership.lastModifiedTimeInMS}, ETag: ${ownership.eTag}`
+        );
+      } catch (err) {
+        log.error(
+          `Error ocuured while claiming ownership for partition: ${ownership.partitionId}`,
+          err
+        );
+      }
+    }
+    return partitionOwnershipArray;
+  }
+
+  /**
+   * Updates the checkpoint in the data store for a partition.
+   *
+   * @param checkpoint The checkpoint.
+   * @return The new eTag on successful update
+   */
+  async updateCheckpoint(checkpoint: Checkpoint): Promise<string> {
+    const blobName = `${checkpoint.eventHubName}/${checkpoint.consumerGroupName}/${checkpoint.partitionId}`;
+    let ownerId;
+    let blob;
+    try {
+      for await (const blobItem of this._containerClient.listBlobsFlat({
```

Comment:
Is the purpose of getting the blob 1st here to do additional validation?

Doing the ownerId validation in `updateCheckpoint` makes the workflow more difficult if someone wanted to use this package for manual checkpointing outside of EventProcessor, and I think the EventProcessor already is protected indirectly with the use of etag. We should prioritize minimizing the number of calls we make when possible. 

Did you run into a use case when testing with EventProcessor where you needed this validation? I'm for leaving it in if it helps `EventProcessor` for now.

Note: Since we are only checking a single blob, I think this would be a case where the BlobClient would be better than calling `listBlobsFlat`, since `listBlobsFlat` can potentially make multiple calls if there are enough partitions being processed.

Reply:
>Is the purpose of getting the blob 1st here to do additional validation?

I wanted to make sure the ownerId matches before updating the checkpoint. I agree with you about avoiding multiple network calls, so I was thinking instead of calling `listBlobsFlat` what about if we the use the below:
```javascript
   const blobClient = this._containerClient.getBlobClient(blobName);
   const properties = await blobClient.getProperties();
   if (properties.metadata!.ownerid !== checkpoint.ownerId) {
      log.error(
        `ownerId: [${checkpoint.ownerId}] doesn't match with stored ownerId: [${properties.metadata!.ownerid}].`
      );
      throw new Error(
        `ownerId: [${checkpoint.ownerId}] doesn't match with stored ownerId: [${properties.metadata!.ownerid}].`
      );
   }
```
This will reduce network latency and will provide better error to the user.

Do you still think that `Doing the ownerId validation in updateCheckpoint makes the workflow more difficult`  ?